shared/chat/inbox/row/small-team/index.js

```diff
@@ -39,23 +39,51 @@ type Props = {
   youNeedToRekey: boolean,
 }
 
+type State = {
+  isHovered: boolean,
+}
+
 const SmallTeamBox = isMobile
-  ? Box
+  ? ClickableBox
   : glamorous(Box)({
       '& .small-team-gear': {display: 'none'},
       ':hover .small-team-gear': {display: 'unset'},
       ':hover .small-team-timestamp': {display: 'none'},
     })
 
-class SmallTeam extends React.PureComponent<Props> {
+class SmallTeam extends React.PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props)
+    this.state = {
+      isHovered: false,
+    }
+  }
+
+  _backgroundColor = () =>
+    // props.backgroundColor should always override hover styles, otherwise, there's a
+    // moment when the conversation is loading that the selected inbox row is styled
+    // with hover styles instead of props.backgroundColor.
+    this.props.isSelected
+      ? this.props.backgroundColor
+      : this.state.isHovered
+        ? globalColors.blue4
+        : this.props.backgroundColor
```

Comment:
Maybe return early if it's selected and use a ternary for `isHovered?

```js
if (this.props.isSelected) return this.props.backgroundColor

return this.state.isHovered ? globalColors.blue4 : this.props.backgroundColor
```

Reply:
I've seen this logic elsewhere in the app (the nested ternary operators). I assume it's to reduce `return`s and brackets and etc. So, I'm not sure if this is a better or just different.

I'm wondering if how I'm approaching this is wrong. I mean, this works, but is there a better way to achieve what I want without even needing this?

Thanks for the feedback, though!

Reply:
yeah no problem! I mean everything is just binary in the end so...

 👾 do ✨ what  🌈 you 🐰 feel 🦄  like 🍠  man! 🥇